pandas/core/indexes/multi.py

```diff
@@ -3426,6 +3426,13 @@ def isin(self, values, level=None):
             else:
                 return np.lib.arraysetops.in1d(level_codes, sought_labels)
 
+    @property
+    def shape(self):
+        """
+        Return a tuple of the shape of the underlying data.
+        """
+        return (len(self),)
```

Comment:
Hmm I realize that this only gives the length on master but maybe should give nlevels as well? @toobaz

Reply:
To be honest, I never used ``Index.shape``... I _guess_ it could return a 2-items tuple, yes, but I can't think of any particularly useful use case, and that would break the (admittedly still unwritten) rule that a 1-level ``MultiIndex`` is API-equivalent to a flat ``Index``.

This said, _if_ we are fine with leaving the current behavior: couldn't the method you wrote just become the default ``Index.shape``? I ask also because if I remember correctly ``MultiIndex`` is not the only type of ``Index`` for which ``._values`` is expensive.

Reply:
@toobaz Yes, I believe this could be moved into `Index.shape` if that's the desired direction. `IntervalIndex.shape` is also overriden in this way.

Reply:
yes pls move this (and I am not sure that we have *any* tests for this .shape is not generally used in index methods).

Reply:
Moved, benchmarks below